pkg/reconciler/shared/tektonconfig/pipeline/pipeline.go

```diff
@@ -20,35 +20,31 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"log"
 	"reflect"
 
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
 
 	op "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
-	operatorv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"knative.dev/pkg/test/logging"
 )
 
-func CreatePipelineCR(ctx context.Context, instance v1alpha1.TektonComponent, client operatorv1alpha1.OperatorV1alpha1Interface) error {
-	configInstance := instance.(*v1alpha1.TektonConfig)
-	if _, err := ensureTektonPipelineExists(ctx, client.TektonPipelines(), configInstance); err != nil {
-		return errors.New(err.Error())
-	}
-	if _, err := waitForTektonPipelineState(ctx, client.TektonPipelines(), v1alpha1.PipelineResourceName,
-		isTektonPipelineReady); err != nil {
-		log.Println("TektonPipeline is not in ready state: ", err)
-		return err
+func EnsureTektonPipelineExists(ctx context.Context, clients op.TektonPipelineInterface, config *v1alpha1.TektonConfig) (*v1alpha1.TektonPipeline, error) {
+	tpCR, err := GetPipeline(ctx, clients, v1alpha1.PipelineResourceName)
+
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			tpCR, err = CreatePipeline(ctx, clients, config)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, v1alpha1.RECONCILE_AGAIN_ERR
+		}
 	}
-	return nil
-}
 
-func ensureTektonPipelineExists(ctx context.Context, clients op.TektonPipelineInterface, config *v1alpha1.TektonConfig) (*v1alpha1.TektonPipeline, error) {
-	tpCR, err := GetPipeline(ctx, clients, v1alpha1.PipelineResourceName)
 	if err == nil {
```

Comment:
```suggestion
	
```
we don't need this explicity `err == nil` check as that is the default code path here.

Reply:
and it will be great if we can move lines `50 updated := flase` to `74 tpCR, err = clients.Update(ctx, tpCR, metav1.UpdateOptions{})` into a function

Reply:
yes I was thinking the same 
So our code would have functions like
```
Get pipeline
Create pipeline
Update pipeline
```